ops/charm.py

```diff
@@ -500,6 +619,7 @@ def restore(self, snapshot: dict) -> None:
                 msg = 'failed loading storage (name={!r}, index={!r}) from snapshot' \
                     .format(storage_name, storage_index)
                 raise RuntimeError(msg)
+            assert storage_location  # type guard
```

Comment:
If optimizations are ever enabled for in a production environment, this assert won't run.  Would we rather have an if-guard and raise maybe?